src/Microsoft.Azure.SignalR.Management/ServiceManager.cs

```diff
@@ -29,7 +37,50 @@ public Task<IServiceHubContext> CreateHubContextAsync(string hubName)
             {
                 case ServiceTransportType.Persistent:
                     {
-                        throw new NotImplementedException();
+                        var loggerFactory = new LoggerFactory();
+                        var endpoint = new ServiceEndpoint(_serviceManagerOptions.ConnectionString);
+                        var serverOptions = new ServiceOptions
+                        {
+                            ConnectionString = _serviceManagerOptions.ConnectionString,
+                            ConnectionCount = ServerConnectionCount,
+                            Endpoints = new ServiceEndpoint[] { endpoint }
+                        };
+                        var options = Options.Create(serverOptions);
+                        var endpointManager = new ServiceEndpointManager(options, loggerFactory);
+                        var provider = endpointManager.GetEndpointProvider(endpoint);
+                        var connectionFactory = new ConnectionFactory(hubName, provider, loggerFactory);
+                        var serviceProtocol = new ServiceProtocol();
+                        var clientConnectionManager = new ClientConnectionManager();
+                        var clientConnectionFactory = new ClientConnectionFactory();
+                        ConnectionDelegate connectionDelegate = connectionContext => Task.CompletedTask;
+                        var serviceConnectionFactory = new ServiceConnectionFactory(serviceProtocol, clientConnectionManager, loggerFactory, connectionDelegate, clientConnectionFactory);
+                        var weakConnectionContainer = new WeakServiceConnectionContainer(serviceConnectionFactory, connectionFactory, ServerConnectionCount, endpoint);
+
+                        var serviceCollection = new ServiceCollection();
+                        serviceCollection.AddSignalRCore();
+
+                        serviceCollection
+                            .AddSingleton(typeof(ILogger<DefaultHubProtocolResolver>), NullLogger<DefaultHubProtocolResolver>.Instance)
+                            .AddSingleton(typeof(HubLifetimeManager<>), typeof(ServiceLifetimeManagerCore<>))
+                            .AddSingleton(typeof(IServiceEndpointManager), typeof(ServiceEndpointManager))
+                            .AddSingleton(typeof(IServiceProtocol), typeof(ServiceProtocol))
+                            .AddSingleton(typeof(IServiceConnectionManager<>), typeof(ServiceConnectionManager<>))
+                            .AddSingleton<IHostedService, HeartBeat>()
+                            .AddSingleton<NegotiateHandler>()
```

Comment:
>NegotiateHandler [](start = 42, length = 16)

I thought it was a simple web socket connection, can we simplify the dependencies?

Reply:
Sounds good. I can simply create one or more service connections to read/write messages.
In this way, I don't need to rely on `Microsoft.Azure.SignalR` any more.
But I need to write a new `HubLifetimeManager`, which is almost the same as `ServiceLifetimeManager`.
But advantages outweigh disadvantages.

---
My purpose is to get an instance of `IHubContext`.
If I simply create a web socket connection, I need to implement `IHubContext`. Then I need to implement `IHubClients` and `IClientProxy` which are used in `IClientProxy` (UserProxy, MultipleUserProxy, GroupProxy, ... etc)